src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,625 @@
+                        @Override
+                        public int compare(File o1, File o2) {
+                            return String.CASE_INSENSITIVE_ORDER.compare(o1.getName(), o2.getName());
+                        }
+                    });
+                    for (File file : listOfFiles) {
+                        result.add(file);
+                    }
+                    return result;
+                case BY_LEXICOGRAPHICAL_DESCENDING:
+                    folder = new File(getPath());
+                    listOfFiles = folder.listFiles(fileFilter);
+                    Arrays.sort(listOfFiles, new Comparator<File>() {
+                        @Override
+                        public int compare(File o1, File o2) {
+                            return String.CASE_INSENSITIVE_ORDER.compare(o2.getName(), o1.getName());
+                        }
+                    });
+                    for (File file : listOfFiles) {
+                        result.add(file);
+                    }
+                    return result;
+
+
+                case BY_LAST_UPDATE_DESCENDING:
+                    folder = new File(getPath());
+                    listOfFiles = folder.listFiles(fileFilter);
+                    Arrays.sort(listOfFiles, new Comparator<File>() {
+                        @Override
+                        public int compare(File o1, File o2) {
+                            return Long.valueOf(o1.lastModified()).compareTo(o2.lastModified());
+                        }
+                    });
+                    for (File file : listOfFiles) {
+                        result.add(file);
+                    }
+                    return result;
+                case BY_LAST_UPDATE_ASCENDING:
+                    folder = new File(getPath());
+                    listOfFiles = folder.listFiles(fileFilter);
+                    Arrays.sort(listOfFiles, new Comparator<File>() {
+                        @Override
+                        public int compare(File o1, File o2) {
+                            return Long.valueOf(o2.lastModified()).compareTo(o1.lastModified());
+                        }
+                    });
+                    for (File file : listOfFiles) {
+                        result.add(file);
+                    }
+                    return result;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String fileInfo(File file) {
+        String divider = "|";
+        long size;
+        String str2;
+
+        if (file.isDirectory()) {
+            size = 0;
+        } else {
+            size = file.length();
+        }
+
+        SimpleDateFormat sdf = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss");
+        String str1 = sdf.format(file.lastModified());
+        if (file.isDirectory()) str2 = file.getName() + "/";
+        else str2 = file.getName();
+
+        return String.format("%-20s %15s %s %n", str1 + divider, size + divider, str2);
```

Comment:
There is no need to additionally add your divider to file creation date or file size.
Just print them in your format: `"%20s | %15s | %s"`